src/service/resource.js

```diff
@@ -953,12 +953,15 @@ export class Resource {
 
     const promise = new Promise((resolve, reject) => {
       Services.vsyncFor(this.hostWin).mutate(() => {
+        let callbackResult;
         try {
-          resolve(this.element.layoutCallback(signal));
+          callbackResult = this.element.layoutCallback(signal);
         } catch (e) {
           reject(e);
         }
+        Promise.resolve(callbackResult).then(resolve, reject);
```

Comment:
This is equivalent, but less bytes:
```suggestion
        resolve(callbackResult);
```

Reply:
It has to be `Promise.resolve().then(resolve)` to give a chance for `abort => reject` to be called in case of aborted signal.